subprojects/gst-plugins-base/gst-libs/gst/gl/egl/gsteglcaps.c

```diff
@@ -0,0 +1,287 @@
+/* GStreamer
+ *  Copyright (C) 2022 Intel Corporation
+ *     Author: Liu Yinhang <yinhang.liu@intel.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public
+ * License along with this library; if not, write to the0
+ * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+/**
+ * SECTION:gsteglcaps
+ * @short_description: EGL modifier
+ * @title: GstEGLCaps
+ * @see_also:
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "gsteglcaps.h"
+#include "gstegl.h"
+
+#include <gst/gl/gstglmemory.h>
+#include <gst/allocators/gstdmabuf.h>
+#include <libdrm/drm_fourcc.h>
+
+static void
+fmts_append (GValue * fmts, const gchar * fmt_str)
+{
+  GValue v = G_VALUE_INIT;
+  g_value_init (&v, G_TYPE_STRING);
+  g_value_set_string (&v, fmt_str);
+
+  gst_value_list_append_value (fmts, &v);
+  g_value_unset (&v);
+}
+
+static void
+mod_fmts_append (GValue * mod_fmts, const gchar * fmt_str, guint64 mod)
+{
+  gchar *mod_fmt_str = g_strdup_printf ("%s:0x%016lx", fmt_str, mod);
+
+  fmts_append (mod_fmts, mod_fmt_str);
+  g_free (mod_fmt_str);
+}
+
+static gboolean
+gst_gl_get_modifiers (GstGLContext * context, guint32 fourcc, GValue * modifiers)
```

Comment:
You can move this function to gsteglimage.c, where all EGL functions are called there.
No matter whether the caps helper functions are needed or not in future(because the frame may provide the helpers in framework), this function is indeed needed.